### PR TITLE
Simplify build_layout logic

### PR DESCRIPTION
**Pull request by t-bltg**
As per https://github.com/JuliaPlots/Plots.jl/blob/3f110d6a6c5bb75d5f02e55b3235595d90ea17ac/src/layouts.jl#L547.

### CODE REVIEW

File: src/layouts.jl [hunk 1]
```diff
@@ -494,34 +494,42 @@ layout_args(huh) = error("unhandled layout type $(typeof(huh)): $huh")
 
 function build_layout(args...)
     layout, n = layout_args(args...)
-    build_layout(layout, n)
+    build_layout(layout, n, Array{Plot}(undef, 0))
```
Comment (t-bltg) [inline src/layouts.jl:497]: Let's see it that doesn't break multiple dispatch.

**Comment by t-bltg** [comment]
I'm merging https://github.com/JuliaPlots/Plots.jl/pull/3598, because https://github.com/JuliaPlots/PlotReferenceImages.jl/pull/100 has been merged.

**Comment by t-bltg** [comment]
We hit https://github.com/JuliaStats/Distributions.jl/issues/1358.

**Review by BeastyBlacksmith — APPROVED** [review]
Whats up with the benchmarks? Can you also re-enable the StatsPlots tests, if they are fixed?

**Comment by t-bltg** [comment]
> Whats up with the benchmarks?

Which benchmarks ?

>  Can you also re-enable the StatsPlots tests, if they are fixed?
That won't work at the moment see the conditions listed here, https://github.com/JuliaPlots/Plots.jl/pull/3611.

**Comment by t-bltg** [comment]
> Can you also re-enable the StatsPlots tests, if they are fixed?
@BeastyBlacksmith there is also the possibility to add `Distributions = "< 0.25.7, ≥ 0.25.9"` to `Project.toml` in github.com/JuliaPlots/StatsPlots.jl to bypass github.com/JuliaStats/Distributions.jl.

**Comment by t-bltg** [comment]
@BeastyBlacksmith, could you explain to me in 2secs  where the benchmark results can be seen ? (not handy to scroll the logs)

I'm not familiar with those: I believe they are used to measure performance regression somehow ?

**Comment by t-bltg** [comment]
> Whats up with the benchmarks?

  Results
  =========
| ID                      | time ratio                   | memory ratio                 |
|-------------------------|------------------------------|------------------------------|
| `["display"]`           | 0.00 (5%) :white_check_mark: | 0.00 (1%) :white_check_mark: |
| `["load_plot_display"]` |                   0.97 (5%)  | 0.00 (1%) :white_check_mark: |
| `["plot"]`              | 0.72 (5%) :white_check_mark: |                   1.00 (1%)  |

**Pull request merged**